homework04/test/examples/Main.hs

```diff
@@ -1,15 +1,37 @@
-module Main
-  ( main
-  )
-  where
-
--- doctest
-import qualified Test.DocTest as DocTest
+module Main where
 
+import Wholemeal
+import Test.Hspec
 
 main :: IO ()
-main =
-  DocTest.doctest
-    [ "-isrc"
-    , "src/Wholemeal.hs"
-    ]
+main = hspec $ do
+
+  describe "fun1'" $
+    it "returns same output of fun1" $
+      fun1' [1,3,5,7] `shouldBe` fun1 [1,3,5,7]
+  
+  describe "fun2'" $
+    it "returns same output of fun2" $
+      fun2' 10 `shouldBe` fun2 10
+  
+  describe "foldTree" $
+    -- Because you can have many implementations of foldTree, you will only test the height of the tree
+    it "returns heigh of a balanced binary tree" $ do      
+      let height (Node x _ _ _) = x
+      (height $ foldTree "ABCDEFGHIJ") `shouldBe` 3
+         
+  describe "xor" $ do
+    it "returns xor from a list with an odd number of True values" $
+      xor [False, True, False] `shouldBe` True
+   
+    it "returns xor from a list with an even number of True values" $
+      xor [False, True, False, False, True] `shouldBe` False
+
+  describe "map'" $
+    it "returns same output of standard map function" $
+      map' (+1) [1,2,3] `shouldBe` map (+1) [1,2,3]
+  
+  describe "sieveSundaram" $
+    it "returns a list of prime numbers up to 2*n + 2" $
```

Comment:
In any implementation, you need to create a list as `[1...n]` in order to remove numbers that are not primes, so if `n` is a negative number or zero, the output will be an empty list.